rskj-core/src/main/java/org/ethereum/facade/EthereumImpl.java

```diff
@@ -54,53 +53,37 @@
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 
-/**
- * @author Roman Mandeleil
- * @since 27.07.2014
- */
-// @Component
 public class EthereumImpl implements Ethereum {
 
     private static final Logger logger = LoggerFactory.getLogger("facade");
     private static final Logger gLogger = LoggerFactory.getLogger("general");
 
-    @Autowired
-    WorldManager worldManager;
-
-    @Autowired
-    AdminInfo adminInfo;
-
-    @Autowired
-    ChannelManager channelManager;
-
-    @Autowired
-    PeerServer peerServer;
-
-    @Autowired
-    ApplicationContext ctx;
-
-    @Autowired
-    ProgramInvokeFactory programInvokeFactory;
-
-    @Autowired
-    PendingState pendingState;
-
-    @Autowired
-    SystemProperties config;
-
-    @Autowired
-    CompositeEthereumListener compositeEthereumListener;
-
-    @Autowired
-    ReceiptStore receiptStore;
+    private WorldManager worldManager;
+    private AdminInfo adminInfo;
+    private ChannelManager channelManager;
+    private PeerServer peerServer;
+    private ProgramInvokeFactory programInvokeFactory;
+    private PendingState pendingState;
+    private SystemProperties config;
+    private CompositeEthereumListener compositeEthereumListener;
+    private ReceiptStore receiptStore;
+    private PeerClient peerClient;
 
     private GasPriceTracker gasPriceTracker = new GasPriceTracker();
 
-    public EthereumImpl() {
-        System.out.println();
+    public EthereumImpl(WorldManager worldManager, AdminInfo adminInfo, ChannelManager channelManager, PeerServer peerServer, ProgramInvokeFactory programInvokeFactory, PendingState pendingState, SystemProperties config, CompositeEthereumListener compositeEthereumListener, ReceiptStore receiptStore, PeerClient peerClient) {
+        this.worldManager = worldManager;
+        this.adminInfo = adminInfo;
+        this.channelManager = channelManager;
+        this.peerServer = peerServer;
+        this.programInvokeFactory = programInvokeFactory;
+        this.pendingState = pendingState;
+        this.config = config;
+        this.compositeEthereumListener = compositeEthereumListener;
+        this.receiptStore = receiptStore;
+        this.peerClient = peerClient;
     }
 
-    @PostConstruct
```

Comment:
Ah, just realized why you called `init` above. Looks good now!

Reply:
Cool